packages/integration-sdk-core/src/data/__tests__/converters.test.ts

```diff
@@ -150,3 +151,56 @@ describe('#truncateEntityPropertyValue', () => {
     );
   });
 });
+
+describe('#parseTimePropertyValue', () => {
+  test('should return undefined', () => {
+    expect(parseTimePropertyValue(null)).toBeUndefined();
+    expect(parseTimePropertyValue(undefined)).toBeUndefined();
+    expect(parseTimePropertyValue('')).toBeUndefined();
+    expect(parseTimePropertyValue(0)).toBeUndefined();
+    expect(parseTimePropertyValue('bargbklselk')).toBeUndefined();
+  });
+  test('should correctly parse Date', () => {
+    expect(parseTimePropertyValue(new Date('2020-10-06T17:41:28Z'))).toBe(
+      1602006088000,
+    );
+    expect(parseTimePropertyValue(new Date('2020-10-06'))).toBe(1601942400000);
+  });
+  test('should correctly parse string', () => {
+    expect(parseTimePropertyValue('2020-10-06T17:41:28Z')).toBe(1602006088000);
+    expect(parseTimePropertyValue('2020-10-06')).toBe(1601942400000);
+    expect(parseTimePropertyValue('2020-10-06T17:41:28.999999999Z')).toBe(
+      1602006088999,
+    );
+    expect(parseTimePropertyValue('2020-10-06T17:41:28.9999+00:00')).toBe(
+      1602006088999,
+    );
+    expect(parseTimePropertyValue('1601942400', 'sec')).toBe(1601942400000);
+    expect(parseTimePropertyValue('1601942400.887', 'sec')).toBe(1601942400887);
+  });
+  test('parsing number without precision', () => {
+    expect(() => parseTimePropertyValue(123)).toThrowError(
+      new Error('Argument precision is required when parsing a number.'),
+    );
+  });
+  test('parsing number', () => {
+    // Seconds
+    expect(parseTimePropertyValue(parseInt('1601942400', 10), 'sec')).toBe(
+      1601942400000,
+    );
+    expect(parseTimePropertyValue(Number(1601942400), 'sec')).toBe(
+      1601942400000,
+    );
+    expect(parseTimePropertyValue(1601942400.99999, 'sec')).toBe(1601942400999);
+    expect(parseTimePropertyValue(1601942400.0083234234, 'sec')).toBe(
+      1601942400008,
+    );
+    // Milliseconds
+    expect(parseTimePropertyValue(1601942400000, 'ms')).toBe(1601942400000);
+    expect(() => parseTimePropertyValue(1601942400.001, 'ms')).toThrowError(
+      new Error(
+        'Argument time must not be a float when precision is set to ms.',
```

Comment:
Strictly speaking I believe the Date() constructor will allow a floating point number to be passed to it, it'll just ignore everything after the decimal.  Feels more like a time to warn instead of throwing an error.